### PR TITLE
implement initial kind checking for sgir

### PR DESCRIPTION
**Pull request by aatxe**

### CODE REVIEW

File: src/sgir.rs [hunk 1]
```diff
@@ -19,15 +19,39 @@ pub struct TypeBinding {
...
+fn compute_kinds(kenv: &KindEnv, typ: Type) -> Kind {
+    match typ {
+        Type::Variable(id) => kenv[&id].clone(),
+        Type::Instantiate { typ, arguments } => match *typ {
+            Type::ForAll { parameters, typ } => {
+                let mut extended_kenv = kenv.clone();
+                extended_kenv.extend(parameters.into_iter()
+                                      .zip(arguments.into_iter())
+                                      .map(|(param, arg)| (param.id, compute_kinds(kenv, arg))));
+                // lol i should actually check that the arguments match
```
Comment (jaykru) [inline src/sgir.rs]: i agree

Comment (aatxe) [reply src/sgir.rs]: it's a draft for a reason! 😭

File: src/sgir.rs [hunk 2]
```diff
@@ -11,44 +12,96 @@ pub enum Kind {
...
         parameters: Vec<TypeBinding>,
         typ: Box<Type>,
     },
+    /// type instantiation, e.g. T<U...>
     Instantiate {
         typ: Box<Type>,
         arguments: Vec<Type>,
     },
+
+    // TODO: existential quantification
```
Comment (jaykru) [inline src/sgir.rs]: 😭

**Review by jaykru — APPROVED** [review]

File: src/sgir.rs (hunk 2 above)
Comment (aatxe) [reply src/sgir.rs]: i'll get to it in the next one

**Pull request merged**